tests/system/test_dbapi.py

```diff
@@ -643,6 +863,17 @@ def test_read_only(self):
         ReadOnly transactions.
         """
 
+        self._conn.read_only = True
+        self._cursor.execute("SELECT * FROM contacts")
+        self._conn.commit()
+        assert self._cursor.fetchall() == []
```

Comment:
nit: This should be before the `commit()` call. Using a transaction after it has been committed is not (always) safe.